packages/prettier-plugin-java/src/printers/printer-utils.js

```diff
@@ -443,6 +443,67 @@ function isShiftOperator(tokens, index) {
   return "none";
 }
 
+function retrieveNodesToken(ctx) {
+  const tokens = retrieveNodesTokenRec(ctx);
+  tokens.sort((token1, token2) => {
+    return token1.startOffset - token2.startOffset;
+  });
+  return tokens;
+}
+
+function retrieveNodesTokenRec(ctx) {
+  const tokens = [];
+  if (ctx && ctx.hasOwnProperty("image") && ctx.tokenType) {
+    if (ctx.leadingComments) {
+      tokens.push(...ctx.leadingComments);
+    }
+    tokens.push(ctx);
+    if (ctx.trailingComments) {
+      tokens.push(...ctx.trailingComments);
+    }
+    return tokens;
+  }
+  Object.keys(ctx.children).forEach(child => {
+    ctx.children[child].forEach(subctx => {
+      tokens.push(...retrieveNodesTokenRec(subctx));
+    });
+  });
+  return tokens;
+}
+
+function buildOriginalText(tokens) {
+  if (tokens.length == 0) {
+    ("");
+  }
+  if (tokens.length == 1) {
+    return dedent(concat([tokens[0]]));
+  }
+  const originalText = [];
+  let lastToken = tokens[0];
+  let startLine;
+  let space;
+  for (let i = 0; i < tokens.length; i++) {
+    startLine = tokens[i].startLine;
+    while (startLine - lastToken.startLine > 0) {
+      originalText.push(hardline);
+      startLine--;
+    }
+    if (tokens[i].startLine - lastToken.startLine > 0) {
+      space = tokens[i].startColumn - 1;
+    } else {
+      space = tokens[i].startColumn - lastToken.endColumn - 1;
+    }
+    while (space > 0) {
+      originalText.push(" ");
```

Comment:
does not this assume that each space is a regular space?
What if the space was a tab?

Reply:
Oh right I have not thought about it since my configured my IDE to consider tab as spaces.

Reply:
if we chop of the relevant part from the original string this issue would be resolved.

Reply:
Yup ! I implemented it and it is working pretty well 👍